src/states/elder/mod.rs

```diff
@@ -1647,26 +1647,29 @@ impl Approved for Elder {
 
         let pub_id = details.pub_id;
 
-        // We need proof that is valid for both the relocating node and the target section. To
-        // construct such proof, we create one proof for the relocating node and one for the target
-        // section and then take the longer of the two. This works because the longer proof is a
-        // superset of the shorter one. We need to do this because in rare cases, the relocating
-        // node might be lagging behind the target section in the knowledge of the source section.
-        let proof = {
-            let proof_for_source = self.chain.prove(&Authority::Node(*details.pub_id.name()));
-            let proof_for_target = self.chain.prove(&Authority::Section(details.destination));
-
-            if proof_for_source.blocks_len() > proof_for_target.blocks_len() {
-                proof_for_source
-            } else {
-                proof_for_target
-            }
-        };
+        // Do not send the message to ourselves.
```

Comment:
I am not clear this patch makes complete sense.
Other nodes still have the ConnectionInfo, and will also try to send that message. My understanding is that if the node already received one from one node, the section node will try to connect to send that message leading the the same outcome as when node sends to itself.

Shouldn't we check it is a bootstrap node that is connected before sending boostrap request instead?

Reply:
> I am not clear this patch makes complete sense.
The main issue is that we currently send `BootstrapRequest` to whoever gets connected to us first. If it happens to be us (how that can happen is described in the commit message), then we are stuck, because we are not going to respond to it. If it is anyone else, even someone from our old section, then in the worst case we get `Rebootstrap` response and proceed normally. So this patch is just a minimal fix to make sure we never send a message to ourselves to prevent this problem.

A more robust fix would be to make sure that we only send `BootstrapRequest` to someone we explicitly connected to via `bootstrap` or `connect_to` and ignore the rest. So basically what you suggest in the last paragraph. Maybe we can use the `nodes_to_await` collection for that somehow. I would handle it in a separate PR as I'm trying to minimize the amount of changes going into this already huge PR. Is that OK with you?

Reply:
If you think that is ok, I'm happy re-think it later.